s3gw/allow_deny.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+import (
+	"fmt"
+	"net"
+	"strings"
+)
+
+// AllowDeny represents address based access control like hosts.allow
+type AllowDeny struct {
+	allowed []*net.IPNet
+	denied  []*net.IPNet
+}
+
+func parseCIDRs(cidrs, name string) ([]*net.IPNet, error) {
+	var nets []*net.IPNet
+
+	for _, s := range strings.Split(cidrs, ",") {
+		s = strings.Trim(s, " ")
+		if s == "" {
+			continue
+		}
+		_, n, err := net.ParseCIDR(s)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse %s CIDR: %w", name, err)
+		}
+		nets = append(nets, n)
+	}
+
+	return nets, nil
+}
+
+// ParseAllowDeny parses ACL. Note: it only accepts CIDR. (hostnames are not supported)
+func ParseAllowDeny(allowed, denied string) (*AllowDeny, error) {
```

Comment:
`ParseAllowDeny` doesn't need to be exported since it is only called from the same package `main`. Or is it more appropriate to separate the package from `main`?

Reply:
No specific reason to capitalize.

By the way, what about the other symbols that start with a uppercase letter?